python/dex/__init__.py

```diff
@@ -52,15 +52,35 @@ def eval(expr: str, module=prelude, _env=None):
   result = api.evalExpr(_env, api.as_cstr(expr))
   if not result:
     api.raise_from_dex()
-  return Atom(result, module)
+  return Atom._from_ptr(result, module)
 
 
 class Atom:
   __slots__ = ('__weakref__', '_as_parameter_', 'module')
 
-  def __init__(self, ptr, module):
+  def __init__(self, value):
+    catom = api.CAtom()
+    if isinstance(value, int):
+      catom.tag = 0
+      catom.value.tag = 1
+      catom.value.value = ctypes.c_int(value)
+    elif isinstance(value, float):
+      catom.tag = 0
+      catom.value.tag = 4
+      catom.value.value = ctypes.c_float(value)
+    else:
+      raise ValueError("Can't convert given value to a Dex Atom")
+    self.module = prelude
```

Comment:
Shouldn't this be an optional argument, that is defaulted to `prelude` ?
Or is it fine because as a literal it is harmless to add it to the prelude and every other module is sure to have access to things from `prelude` ?

Reply:
Yeah the `module` should be a set of bindings that includes all definitions of free variables appearing in the atom. But if the atom is a literal then it has no free variables, so any module would do. But a prelude seems like a reasonable default.